clash-cores/src/Clash/Cores/Xilinx/Ethernet/Gmii.hs

```diff
@@ -0,0 +1,136 @@
+  Signal gmiiTx125 Gmii ->
+  -- | Output record, see 'BridgeOutput'.
+  BridgeOutput gmiiRx125 sgmii625
+gmiiSgmiiBridge refClk refRst cLinkSpeed signalDetect bridgeConfig anConfig anRestart lvdsIn gmiiTx = BridgeOutput{..}
+ where
+  ( lvdsOutP
+    , lvdsOutN
+    , gmiiRxD
+    , gmiiRxDv
+    , gmiiRxEr
+    , bridgeClk125
+    , activeHighRxRst
+    , fmap fromStatusVector -> bridgeStatusVector
+    ) = gmiiSgmiiBridgeBb clockP clockN activeHighRefRst (getSpeed100 <$> cLinkSpeed)
+      (getSpeed10_100 <$> cLinkSpeed) signalDetect (toConfigVector <$> bridgeConfig)
+      anConfig anRestart lvdsInP lvdsInN gmiiTxD gmiiTxEn gmiiTxEr
+
+  activeHighRefRst = unsafeToActiveHigh refRst
+  bridgeRst125 = unsafeFromActiveHigh activeHighRxRst
+  bridgeLvdsOut = Lvds <$> lvdsOutP <*> lvdsOutN
+  (DiffClock clockP clockN) = refClk
+  (lvdsInP, lvdsInN) = (fmap lvdsP lvdsIn, fmap lvdsN lvdsIn)
+  (gmiiTxD, gmiiTxEn, gmiiTxEr) = unbundle $ fmap (\Gmii{..} -> (gmiiData, gmiiValid, gmiiError)) gmiiTx
+  bridgeGmiiRx = Gmii <$> gmiiRxD <*> gmiiRxDv <*> gmiiRxEr
+
+-- | Black box for the GMII to SGMII bridge.
+gmiiSgmiiBridgeBb ::
+  -- | P channel of the reference clock coming from the Phy.
+  Clock sgmii625 ->
+  -- | N channel of the reference clock coming from the Phy.
+  ClockN sgmii625 ->
+  -- | Asynchronous active high reset for the bridge.
+  Signal asyncDom Bool->
+  -- | Whether the link should be configured for 100 Mbit/s.
+  Signal gmiiRx125 Speed_100 ->
+  -- | Whether the link should be configured for 10 or 100 Mbit/s.
+  Signal gmiiRx125 Speed_10_100 ->
+  -- | Signal detect from the Phy.
+  Signal asyncDom SignalDetect ->
+  -- | Configuration for the bridge.
+  Signal gmiiRx125 ConfigVector ->
+  -- | Auto negotiation configuration for the bridge.
+  Signal gmiiRx125 AutoNegConfig ->
+
+  -- | Restart auto negotiation.
+  Signal gmiiRx125 RestartAutoNeg ->
+  -- | P channel of the LVDS input from the Phy.
+  Signal sgmii625 Bit ->
+  -- | N channel of the LVDS input from the Phy.
+  Signal sgmii625 Bit ->
+  -- | GMII data input from the MAC.
+  Signal gmiiTx125 (BitVector 8) ->
+  -- | GMII valid input from the MAC.
+  Signal gmiiTx125 Bit ->
+  -- | GMII error input from the MAC.
+  Signal gmiiTx125 Bit ->
+  -- |
+  -- 1. P channel of the LVDS output to the Phy.
+  -- 2. N channel of the LVDS output to the Phy.
+  -- 3. GMII data output to the MAC.
+  -- 4. GMII valid output to the MAC.
+  -- 5. GMII error output to the MAC.
+  -- 6. Clock output for the 125 MHz domain.
+  -- 7. Active high reset output for the 125 MHz domain.
+  -- 8. Status vector.
+  ( Signal sgmii625 Bit
+  , Signal sgmii625 Bit
+  , Signal gmiiRx125 (BitVector 8)
+  , Signal gmiiRx125 Bit
+  , Signal gmiiRx125 Bit
+  , Clock gmiiRx125
+  , Signal gmiiRx125 Bool
+  , Signal gmiiRx125 StatusVector
+  )
+
+gmiiSgmiiBridgeBb !_ !_  !_ !_ !_ !_ !_ !_ !_ !_ !_ !_ !_ !_ =
+  (undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined)
```

Comment:
Why not `deepErrorX`?

Reply:
One of the results is a `Clock`, added a comment.